File: PreUtil/src/output/python/easy/ReverseLinkedList.py
#https://leetcode.com/problems/reverse-linked-list/
#Given the head of a singly linked list, reverse the list, and return the reversed list.


# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """ 
        if head==None:
            return head
        cur=head
        prev=None
        while cur!=None:
            next=cur.next
            cur.next=prev
            prev=cur
            cur=next
        return prev

    

File: PreUtil/src/output/javascript/easy/ReverseLinkedList.js
/**https://leetcode.com/problems/reverse-linked-list/ */
//Given the head of a singly linked list, reverse the list, and return the reversed list.


/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    let prev = null;
    let current = head;
    while(current != null){
        let next = current.next;
        current.next = prev;
        prev = current;
        current = next;
    }
    return prev;
    
};

    